modules/broker-core/src/main/java/io/ballerina/messaging/broker/core/store/dao/impl/MessageDaoImpl.java

```diff
@@ -51,57 +53,82 @@
 
     @Override
     public void persist(TransactionData transactionData) throws BrokerException {
-        crudOperationsDao.transaction(connection -> {
-            crudOperationsDao.storeMessages(connection, transactionData.getEnqueueMessages());
-            crudOperationsDao.detachFromQueue(connection, transactionData.getDetachMessageMap());
-            crudOperationsDao.delete(connection, transactionData.getDeletableMessage());
-        });
+        try {
+            crudOperationsDao.transaction(connection -> {
+                crudOperationsDao.storeMessages(connection, transactionData.getEnqueueMessages());
+                crudOperationsDao.detachFromQueue(connection, transactionData.getDetachMessageMap());
+                crudOperationsDao.delete(connection, transactionData.getDeletableMessage());
+            });
+        } catch (DaoException e) {
```

Comment:
What if we throw the DaoException itself from the DAO layer?. That would make sense too IMO. And we can get rid of this added Exception handling.

Reply:
Fixed in 4859054